astroid/node_classes.py

```diff
@@ -3252,6 +3285,35 @@ def get_children(self):
         yield from self.body
         yield from self.orelse
 
+    def _get_assign_nodes(self):
```

Comment:
A lot of these functions are copy-pasted between these classes. Do you think we could use a Mixin class to prevent code duplication?